src_c/surface.c

```diff
@@ -2105,6 +2109,193 @@ surf_blits(pgSurfaceObject *self, PyObject *args, PyObject *keywds)
     return RAISE(PyExc_TypeError, "Unknown error");
 }
 
+#define UBLITS_ERR_TUPLE_REQUIRED 11
+
+static PyObject *
+surf_ublits(pgSurfaceObject *self, PyObject *args, PyObject *keywds)
+{
+    SDL_Surface *src, *dest = pgSurface_AsSurface(self);
+    SDL_Rect *src_rect, temp;
+    temp.x = 0;
+    temp.y = 0;
+    PyObject *srcobject = NULL, *argpos = NULL;
+    int result;
+    SDL_Rect dest_rect;
+    int sx, sy;
+    int the_args = 0;
+
+    PyObject *blitsequence;
+    PyObject *iterator = NULL;
+    PyObject *item = NULL;
+    PyObject *special_flags = NULL;
+    PyObject *ret = NULL;
+    PyObject *retrect = NULL;
+    Py_ssize_t itemlength;
+    int doreturn = 1;
+    int bliterrornum = 0;
+    static char *kwids[] = {"blit_sequence", "special_flags", "doreturn",
```

Comment:
I think it's better to have the function return something at all times instead of having a ``doreturn`` kwarg

Reply:
Yeah this is something blits() implements and i thought keeping the same use woud be the way. If it's there is because you can use the returned rect list later for other things. We can either remove this functionality entirely or keep it. I'm not sure...

Reply:
> 🥳 Nice PR, found one problem though.

and thank you, first ever and i've never seen anything about the Python ABI before, so this is just messing up with stuff for many hours :)

Reply:
Oh, I didn't know that. Then I think we should keep it. I'll keep this unresolved incase there's opposition.

Reply:
>I think it's better to have the function return something at all times instead of having a `doreturn` kwarg even if the performance is slightly boosted without the return. If we want to keep it, we should rename it to `return_rects`

Yes but wouldn't that be the same as now? Do you mean that we would need to create something like an empty list on every call and return that? but in that case you'd still need some flag to decide whether to fill up the list or not.

Reply:
The renaming of `doreturn` to `return_rects` alone would be nice for readability reasons tho. The problem is that `doreturn` is used for `blits()` aswell, so you'd have to change that too be level. Thats not for this PR tho.